diarymanagement-code-r273/tbaclient/tags/final.rc1/tbaclient/src/TBA/Client/GUI/SettingsDialog.java

```diff
@@ -0,0 +1,751 @@
+
+                temp = in.readLine(); //gestures
+
+                temp = in.readLine(); //logon
+                names[0] = temp.substring(temp.indexOf("=")+1);
+
+                temp = in.readLine(); //add new entry
+                names[1] = temp.substring(temp.indexOf("=")+1);
+
+                temp = in.readLine(); //settings
+                names[2] = temp.substring(temp.indexOf("=")+1);
+
+                options.add(names[comboName.getSelectedIndex()]);
+                fillComboMethods();
+
+                temp = in.readLine(); //connection
+
+                temp = in.readLine(); //user
+                user = temp.substring(temp.indexOf("=")+1);
+                txtUser.setText(user);
+
+                temp = in.readLine(); //server
+                server = temp.substring(temp.indexOf("=")+1);
+                txtServer.setText(server);
+
+                temp = in.readLine(); //port
+                temp = temp.substring(temp.indexOf("=")+1);
+                port = Integer.parseInt(temp);
+                txtPort.setText(temp);
+
+                in.close();
+
+            } catch (IOException e) {
+                System.out.println(e);
+            }
+        }
+    }
+
+    /**
+     * Getter for the port in the settings dialougue, this value is subservient
+     * to its commandline counterpart - eg if there is a port provided in the
+     * commandline, the main program will choose that over this
+     * @return port field to be retrieved
+     */
+    public int getPort() {
+        return port;
+    }
+
+    /**
+     * Setter for the port in the settings dialogue
+     * @param port - new port to set to
+     */
+    public void setPort(int port) {
+        this.port = port;
+    }
+
+    /**
+     * Getter for the server field in the settings dialogue, this value is subservient
+     * to its commandline counterpart - eg if there is a server provided in the
+     * commandline, the main program will choose that over this
+     * @return server field to be retrieved
+     */
+    public String getServer() {
+        return server;
+    }
+
+    /**
+     * Setter fo the server in the settings dialogue
+     * @param server - new server to set to
+     */
+    public void setServer(String server) {
+        this.server = server;
+    }
+
+    /**
+     * Getter for username if remembered by settings dialogue
+     * @return user field
+     */
+    public String getUser() {
+        return user;
+    }
+
+    /**
+     * Setter of the username if remembered by settings dialogue
+     * @param user - to set to
+     */
+    public void setUser(String user) {
+        this.user = user;
+    }
+
+    /**
+     * This method writes the values to file, and then recalls LoadSettings()
+     * to make sure.
+     */
+    public void Update()
+    {
+
+        try{
+            FileWriter fstream = new FileWriter("settings.ini");
+            BufferedWriter out = new BufferedWriter(fstream);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *out*. The resource is referenced in statements at the following lines: 638, 643, 644, 645, 646, 647, 648, 649, 650, 652, 653, 654, 655, 656, 657, 658, 659, 662, 663, 669, 672, 673, 674, 675, 676. The resource closure statement is at line: 680. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by BufferedWriter.newLine. Either a) close *out* in a try-finally block or b) close the resource by declaring *out* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).